### PR TITLE
Implement Scroll API

### PR DESCRIPTION
**Pull request by hpetru**

### CODE REVIEW

**Comment by solnic** [comment]
Hey thanks for this PR. I'm sorry but I missed it when you opened it :( We're having an issue with the build but once master is green again, could you rebase this?

File: lib/rom/elasticsearch/dataset.rb [hunk 1]
```diff
@@ -276,11 +287,39 @@ def call
...
 
+      # @api private
+      def scroll_response
+        Enumerator.new do |y|
+          current_response = response
+
+          loop do
+            results = current_response.fetch('hits').fetch('hits')
+
+            if results.empty?
```
Comment (v-kolesnikov) [inline lib/rom/elasticsearch/dataset.rb]:
```suggestion
            break if results.empty?
```

File: lib/rom/elasticsearch/dataset.rb [hunk 2]
```diff
@@ -276,11 +287,39 @@ def call
...
+        elsif params[:scroll]
+          scroll_response
         else
           response.fetch('hits').fetch('hits')
         end
       end
 
+      # @api private
+      def scroll_response
+        Enumerator.new do |y|
```
Comment (v-kolesnikov) [inline lib/rom/elasticsearch/dataset.rb]:
```suggestion
        Enumerator.new do |yielder|
```

**Comment by v-kolesnikov** [comment]
Looks nice! I only think about a separate module like existing `QueryMethods` or even a relation's plugin `scroll`. WDYT guys?

**Comment by hpetru** [comment]
@v-kolesnikov sounds like a good idea, just extracted the scroll stuff from dataset

**Pull request merged**